packages/kit/src/runtime/client/router.js

```diff
@@ -258,6 +269,9 @@ export class Router {
 		if (!allow_navigation) return;
 
 		if (this.enabled && this.owns(url)) {
+			if (!replaceState) {
+				state['sveltekit:index'] = ++this.current_history_index;
+			}
```

Comment:
I forgot something.
If `replaceState` is `true`, `state['sveltekit:index']` need to be set to the current history index.

```js
    state['sveltekit:index'] = replaceState ? this.current_history_index : ++this.current_history_index;
```

Reply:
gotcha!